File: 182/solution.hs
main = print $ sum [e | e <- [2..phi-1], gcd e phi == 1, (gcd (e-1) (p-1) +1)*(gcd (e-1) (q-1) + 1) == 9]
    where p = 1009
          q = 3643
          phi = (p-1)*(q-1)

File: 182/solution.go
package main

import "fmt"
import "github.com/roessland/gopkg/mathutil"

func main() {
	p := int64(1009)
	q := int64(3643)
	phi := (p - 1) * (q - 1)

	sum := int64(0)
	for e := int64(2); e < phi; e++ {
		if mathutil.GCD(e, phi) != 1 {
			continue
		}
		numUnconcealed := (mathutil.GCD(e-1, p-1) + 1) * (mathutil.GCD(e-1, q-1) + 1)
		if numUnconcealed == 9 {
			sum += e
		}
	}
	fmt.Printf("%v\n", sum)
}
